sw/otbn/code-snippets/barrett384.S

```diff
@@ -0,0 +1,266 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ *   Constant time 384 bit modular multiplication based on Barrett reduction
+ *   algorithm.
+ *
+ *   Provides a generic modular multiplication kernel for 384 bit wide numbers
+ *   without requiring a special form of the modulus.
+ */
+
+.section .text
+
+/**
+ * Unrolled 768=384x384 bit multiplication.
+ *
+ * Returns c = a x b.
+ *
+ * Flags: Last instruction performs a dummy addition on the highest limb of c.
+ * Therefore the routine is left with the flags being set based on this
+ * operation:
+ *    - M: MSb of the result (and the highest limb of the result)
+ *    - L: LSb of the highest limb of result
+ *    - Z: Indicates if highest limb of result is zero
+ *    - C: Never set
+ *
+ * @param[in] [w9, w8]: a first operand, max. length 384 bit.
+ * @param[in] [w11, w10]: b, second operand, max. length 384 bit.
+ * @param[in] w31: all-zero.
+ * @param[out] [w18, w17, w16]: c, result, max. length 768 bit.
+ *
+ * Clobbered registers: w19, w20
+ * Clobbered flag groups: standard
```

Comment:
What do you mean by "standard" here? Is it flag group 0?

Reply:
Yes, I mean't FG0. You're right, we stopped using the terms "standard" and "extension" for this. Updated the PR